src/SymSpell.cpp

```diff
@@ -627,7 +627,7 @@ vector<SuggestItem> SymSpell::Lookup(xstring input, Verbosity verbosity, int max
 		// \w Alphanumeric characters (including non-latin characters, umlaut characters and digits) plus "_" 
 		// \d Digits
 		// Compatible with non-latin characters, does not split words at apostrophes
-		xregex r(XL("['’\\w-\\[_\\]]+"));
+		xregex r(XL("['’\\w\\[_\\]]+"));
```

Comment:
Thanks for your contribution but is there any problem with the hyphen here?

Reply:
The regex seems to be invalid. Please refer the screenshot.
[Imgur](https://i.imgur.com/Yt5kBtG.png)

Reply:
So usually hyphen can represent range operator or just literally a hyphen, some regex engines are able to differentiate them, some aren't. In this case the C++ engine can understood that we want a hyphen there, not a range. And apparently the one you are testing with (PCRE2) do not support this format (you can try the PCRE or Golang flavor on regex101 and see they can parse it just fine).

So it's not a bug and removing the hyphen will potentially make Symspell split words incorrectly so you need to revert that change or, if you prefer, replace it with the following
 `xregex r(XL("['’\\w\\-\\[_\\]]+"));`
and then I can accept your commit. 

Also, if you wanted to check the to see the regexes are correct or not using different engines (like regex101), please at least replace "\\" with "\" so the other engines can understand it correctly. The double slash is because the regex went through 2 escape processes (c-string escape first and then regex)

Reply:
Got it. Thanks for the explanation!